lib/ModelMixins/TableMixin.ts

```diff
@@ -692,24 +702,65 @@ function TableMixin<T extends Constructor<Model<TableTraits>>>(Base: T) {
                     feature.properties[regionType.regionProp].toLowerCase()
-                  )
-                );
+                  ) ?? [];
+
+                const filteredRegionId = filterRows(regionIds);
 
-                let d: JsonObject | null = isDefined(regionId)
-                  ? this.getRowValues(regionId)
+                let d: JsonObject | null = isDefined(filteredRegionId)
+                  ? this.getRowValues(filteredRegionId)
                   : null;
 
                 if (d === null) return;
 
-                return this.featureInfoFromFeature(
-                  regionType,
-                  // Preserve values from d and insert feature properties after entries from d
-                  Object.assign({}, d, feature.properties, d),
-                  feature.properties[regionType.uniqueIdProp]
-                );
+                // Preserve values from d and insert feature properties after entries from d
+                const featureData = Object.assign({}, d, feature.properties, d);
+
+                const featureInfo = new ImageryLayerFeatureInfo();
+                if (isDefined(regionType.nameProp)) {
+                  featureInfo.name = featureData[regionType.nameProp] as string;
+                }
+
+                featureData.id = feature.properties[regionType.uniqueIdProp];
+                featureInfo.properties = featureData;
+
+                featureInfo.configureDescriptionFromProperties(featureData);
+                featureInfo.configureNameFromProperties(featureData);
+
+                // If time-series region-mapping - show timeseries chart
+                if (
+                  !isDefined(featureData._terria_getChartDetails) &&
+                  this.discreteTimes &&
+                  this.discreteTimes.length > 1 &&
+                  this.activeTableStyle.timeColumn &&
+                  this.activeTableStyle.colorColumn?.type ===
+                    TableColumnType.scalar &&
+                  Array.isArray(regionIds) &&
+                  regionIds.length > 1
+                ) {
+                  const chartColumns = [
+                    this.activeTableStyle.timeColumn,
+                    this.activeTableStyle.colorColumn
+                  ];
+                  featureInfo.properties._terria_getChartDetails = () => ({
+                    title: this.activeTableStyle.colorColumn?.title,
+                    xName: this.activeTableStyle.timeColumn?.title,
+                    yName: this.activeTableStyle.colorColumn?.title,
+                    units: chartColumns.map(column => column.units || ""),
+                    csvData: [
+                      chartColumns.map(col => col!.title).join(","),
+                      ...regionIds.map(i =>
+                        chartColumns
+                          .map(col => col.valueFunctionForType(i))
+                          .join(",")
+                      )
+                    ].join("\n")
```

Comment:
Maybe we could split this off into a new method called `setChartDetailsFunctionForFeatureInfo` to simplify the code a bit?

Reply:
I took some stuff out of `private readonly createRegionMappedImageryLayer` and put it into two new functions `getImageryLayerFilteredRows` and `getImageryLayerFeatureInfo`

Reply:
That's better than my suggestion and looks less dense now!